std/container/array.d

```diff
@@ -428,32 +442,10 @@ if (!is(immutable T == immutable bool))
     this(U)(U[] values...)
     if (isImplicitlyConvertible!(U, T))
     {
-        import core.lifetime : emplace;
-
-        static if (T.sizeof == 1)
-        {
-            const nbytes = values.length;
-        }
-        else
-        {
-            import core.checkedint : mulu;
-            bool overflow;
-            const nbytes = mulu(values.length, T.sizeof, overflow);
-            if (overflow)
-                assert(false, "Overflow");
-        }
-        auto p = cast(T*) enforceMalloc(nbytes);
-        // Before it is added to the gc, initialize the newly allocated memory
-        foreach (i, e; values)
-        {
-            emplace(p + i, e);
-        }
-        static if (hasIndirections!T)
-        {
-            if (p)
-                GC.addRange(p, T.sizeof * values.length);
-        }
-        _data = Data(p[0 .. values.length]);
+        _data = Data(DefaultConstruct.value);
+        _data.reserve(values.length);
+        foreach (ref value; values)
+            _data.insertBack(value);
```

Comment:
`insertBack` has some overhead checking for capacity each time. Since you reserved the right length, you can just `emplace` each element directly like the original code does.

Reply:
Thank you. I have updated accordingly.